Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -421,8 +439,13 @@ void UPSWorldSubsystem::ResetSaveGameData()
 
 	// Re-load a new save game object. Load game from save creates a save file if there is no such
 	LoadGameFromSave();
-	SetCurrentRowByTag(PSCurrentSpotComponentInternal->GetMeshChecked().GetPlayerTag());
-	RefreshProgressionUIElements();
+
+	UPSSpotComponent* SpotComponent = GetSpotComponentByPlayerTag(UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter()->GetPlayerTag());
```

Comment:
never access pointers without checks: `GetLocalPlayerCharacter()->` will crash the game. You can use here ternary operator like
```cpp
const APlayerCharacter* Character = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
UPSSpotComponent* SpotComponent = Character ? GetSpotComponentByPlayerTag(Character) : nullptr;
if (!ensureMsgf(SpotComponent, ...)
...
```

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/40/commits/7da8fc982ed519bee8b3d2d030a3632f2748e64f